x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable_factory.ts

```diff
@@ -66,59 +76,64 @@ export class EmbeddableFactory implements EmbeddableFactoryDefinition {
 
   createFromSavedObject = async (
     savedObjectId: string,
-    input: Partial<EmbeddableInput> & { id: string },
+    input: LensEmbeddableInput,
     parent?: IContainer
   ) => {
+    if (!(input as LensByReferenceInput).savedObjectId) {
```

Comment:
```suggestion
    if (!'savedObjectId' in input) {
```

Reply:
This is a better looking line of code, but it causes a type error on the next line because the type is inferred as `LensByValueInput` which cannot be case back to `LensByReferenceInput`